Modules/_testcapimodule.c

```diff
@@ -6303,7 +6303,8 @@ heapctypesubclasswithfinalizer_init(PyObject *self, PyObject *args, PyObject *kw
 static void
 heapctypesubclasswithfinalizer_finalize(PyObject *self)
 {
-    PyObject *error_type, *error_value, *error_traceback, *m, *oldtype, *newtype;
+    PyObject *error_type, *error_value, *error_traceback, *m;
+    PyObject *oldtype=NULL, *newtype=NULL;
```

Comment:
nit
```C
PyObject *oldtype = NULL, *newtype = NULL;
```

Always put spaces around assignment, Boolean and comparison operators.
from PEP7 :)